torch/_tensor.py

```diff
@@ -763,16 +763,16 @@ def split(self, split_size, dim=0):
             return handle_torch_function(
                 Tensor.split, (self,), self, split_size, dim=dim
             )
-        if isinstance(split_size, int):
-            return super(Tensor, self).split(split_size, dim)
-        elif isinstance(split_size, Tensor):
+        if isinstance(split_size, Tensor):
             try:
                 split_size = int(split_size)
-                return super(Tensor, self).split(split_size, dim)
             except ValueError:
-                return super(Tensor, self).split_with_sizes(split_size, dim)
+                pass
+
+        if isinstance(split_size, int):
+            return torch._VF.split(self, split_size, dim)  # type: ignore[attr-defined]
```

Comment:
Why not directly call `torch.split` here?

Reply:
That wouldn't work because `torch.split` calls `Tensor.split`.

Reply:
Ho we expose split there from c++ and then we override it 5 lines later. Ok I missed that. Sounds good.